apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -871,77 +794,55 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
-    """
+  def fetch_datastream_maximum_storage_retention(realm) do
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
 
     query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(maximum_storage_retention_statement)
-      |> DatabaseQuery.consistency(:quorum)
+      KvStore
+      |> where(group: "realm_config", key: "datastream_maximum_storage_retention")
+      |> select([v], fragment("blobAsInt(?)", v.value))
+      |> put_query_prefix(keyspace_name)
 
-    with {:ok, res} <- DatabaseQuery.call(client, query),
-         ["system.blobasint(value)": maximum_storage_retention] <- DatabaseResult.head(res) do
-      {:ok, maximum_storage_retention}
-    else
-      :empty_dataset ->
-        {:ok, nil}
-
-      %{acc: _, msg: error_message} ->
-        Logger.warning("Database error: #{error_message}.")
-        {:error, :database_error}
-
-      {:error, reason} ->
-        Logger.warning("Failed with reason: #{inspect(reason)}.")
-        {:error, :database_error}
+    with n when is_number(n) or is_nil(n) <- Repo.fetch_one(query, consistency: :quorum) do
+      {:ok, n}
     end
   end
 
-  def fetch_path_expiry(db_client, device_id, interface_descriptor, %Mapping{} = mapping, path)
+  def fetch_path_expiry(realm, device_id, interface_descriptor, %Mapping{} = mapping, path)
       when is_binary(device_id) and is_binary(path) do
-    # TODO: do not hardcode individual_properties here
-    fetch_property_value_statement = """
-    SELECT TTL(datetime_value)
-    FROM individual_properties
-    WHERE device_id=:device_id AND interface_id=:interface_id
-      AND endpoint_id=:endpoint_id AND path=:path
-    """
-
-    fetch_property_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(fetch_property_value_statement)
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.put(:interface_id, interface_descriptor.interface_id)
-      |> DatabaseQuery.put(:endpoint_id, mapping.endpoint_id)
-      |> DatabaseQuery.put(:path, path)
-      |> DatabaseQuery.consistency(:quorum)
-
-    with {:ok, result} <- DatabaseQuery.call(db_client, fetch_property_query),
-         ["ttl(datetime_value)": ttl] when is_integer(ttl) <- DatabaseResult.head(result) do
-      expiry_datetime =
-        DateTime.utc_now()
-        |> DateTime.to_unix()
-        |> :erlang.+(ttl)
-        |> DateTime.from_unix!()
+    %InterfaceDescriptor{interface_id: interface_id} = interface_descriptor
+    %Mapping{endpoint_id: endpoint_id} = mapping
+
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
+
+    q =
+      IndividualProperty
+      |> where(
+        device_id: ^device_id,
+        interface_id: ^interface_id,
+        endpoint_id: ^endpoint_id,
+        path: ^path
+      )
+      |> put_query_prefix(keyspace_name)
+      |> select([p], fragment("TTL(?)", p.reception_timestamp))
```

Comment:
The previous query was selecting the TTL on `datetime_value`, not `reception_timestamp`.
Is this change intended?

Reply:
Yes, because I think that the previous implementation was buggy.
`datetime_value` might not be inserted or updated when a property is set (consider e.g. a `boolean`  endpoint), so the TTL might be out-of date, since TTL is set per-column (see https://docs.datastax.com/en/cql-oss/3.3/cql/cql_using/useExpire.html). On the other hand, `reception_timestamp` is always set when a property is set, therefore it can be relied upon when checking the TTL.